aten/src/ATen/native/Pooling.cpp

```diff
@@ -111,6 +117,46 @@ std::tuple<Tensor, Tensor> max_pool1d_with_indices(
   return std::make_tuple(output, indices);
 }
 
+#if AT_MKLDNN_ENABLED()
+static bool use_mkldnn_maxpool(const Tensor& input, IntArrayRef dilation, bool is_3d) {
+  if (!at::globalContext().userEnabledMkldnn()) {
+    return false;
+  }
+  if (input.is_mkldnn()) {
+    return true;
+  }
+  if (input.sym_numel() <= 1) {
+    return false;
+  }
+  if (input.dim() != (is_3d ? 5 : 4)) {
+    return false;
+  }
+  // Does not support dilation case for now
+  // TODO: Add support for dilation
+  if (!std::all_of(dilation.cbegin(), dilation.cend(), [](int64_t i) {
+        return 1 == i;
+      })) {
+    return false;
+  }
+  if (!((GradMode::is_enabled() && input.requires_grad()) ||
+        input._fw_grad(/*level */ 0).defined())) {
+    bool is_channels_last = is_3d
+        ? (input.suggest_memory_format() == at::MemoryFormat::ChannelsLast3d)
+        : (input.suggest_memory_format() == at::MemoryFormat::ChannelsLast);
+    if (is_3d) {
+      return input.device().is_cpu() && is_channels_last &&
+          (((input.scalar_type() == kBFloat16) && mkldnn_bf16_device_check()) ||
+           ((input.scalar_type() == kHalf) && mkldnn_fp16_device_check()));
+    } else {
+      return input.device().is_cpu() &&
+          (((input.scalar_type() == kBFloat16) && mkldnn_bf16_device_check()) ||
+           ((input.scalar_type() == kHalf) && mkldnn_fp16_device_check()));
```

Comment:
Can we move the reusable code into a helper function? I feels it may also be used somewhere else.